latch_cli/services/local_dev.py

```diff
@@ -0,0 +1,319 @@
+
+        init_local_dev_data = init_local_dev_resp.json()
+        access_key = init_local_dev_data["docker"]["tmp_access_key"]
+        secret_key = init_local_dev_data["docker"]["tmp_secret_key"]
+        session_token = init_local_dev_data["docker"]["tmp_session_token"]
+        port = init_local_dev_data["port"]
+
+        # ecr setup
+        ws_id = current_workspace()
+        if int(ws_id) < 10:
+            ws_id = f"x{ws_id}"
+
+        try:
+            ecr_client = boto3.session.Session(
+                aws_access_key_id=access_key,
+                aws_secret_access_key=secret_key,
+                aws_session_token=session_token,
+                region_name="us-west-2",
+            ).client("ecr")
+
+            ecr_auth = ecr_client.get_authorization_token()
+            docker_access_token = ecr_auth["authorizationData"][0]["authorizationToken"]
+
+        except Exception as err:
+            raise ValueError(f"unable to retrieve an ecr login token") from err
+
+        try:
+            paginator = ecr_client.get_paginator("describe_images")
+            image_name = f"{ws_id}_{pkg_root.name}"
+            iterator = paginator.paginate(repositoryName=image_name)
+            filter_iterator = iterator.search(
+                "sort_by(imageDetails, &to_string(imagePushedAt))[-1].imageTags"
+            )
+            latest = next(filter_iterator)
+            image_name_tagged = f"{config.dkr_repo}/{image_name}:{latest}"
+        except:
+            raise ValueError(
+                "This workflow hasn't been registered yet. "
+                "Please register at least once to use this feature."
+            )
+
+        async with asyncssh.connect(
+            centromere_ip, username=centromere_username
+        ) as ssh_conn:
+            await aioconsole.aprint("Copying your local changes... ", end="")
+            await copy_files(ssh_conn, pkg_root)
+            await aioconsole.aprint("Done.")
+
+            async for ws in websockets.connect(
+                f"ws://{centromere_ip}:{port}/ws",
+                close_timeout=0,
+                extra_headers=headers,
+            ):
+                try:
+                    await aioconsole.aprint(
+                        "Successfully connected to remote instance."
+                    )
+                    try:
+                        await send_message(ws, docker_access_token)
+                        await send_message(ws, image_name_tagged)
+
+                        await aioconsole.aprint(
+                            f"Pulling {image_name}, this will only take a moment... "
+                        )
+                        await get_messages(ws, show_output=False)
+                        await get_messages(ws, show_output=False)
+                        await aioconsole.aprint("Image successfully pulled.\n")
+
+                        while True:
+                            with session.input.raw_mode():
```

Comment:
why?

Reply:
so that i can catch ctrl d as EOFError instead of it causing a crash, its given in the prompt-toolkit docs